documentation/modules/cruise-control/proc-generating-rebalance-proposals.adoc

```diff
@@ -0,0 +1,124 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='proc-generating-optimization-proposals-{context}']
+= Generating optimization proposals 
+
+When you create or update a `KafkaRebalance` resource, Cruise Control generates an xref:con-optimization-proposals-{context}[optimization proposal] for the Kafka cluster based on the configured xref:con-optimization-goals-{context}[optimization goals].
+
+You can then analyze the summary information in the optimization proposal and decide whether to implement it. 
+
+.Prerequisites
+
+* You have xref:proc-deploying-cruise-control-{context}[deployed Cruise Control] to your {ProductName} cluster.
+
+* You have configured xref:con-optimization-goals-{context}[optimization goals].
+
+.Procedure
+
+. Create a `KafkaRebalance` resource:
+
+.. To use the _default optimization goals_ defined in the `Kafka` resource, leave the `spec` property empty:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec: {}
+----
+
+.. To set _user-provided optimization goals_ instead of using the default goals, edit the `goals` property and enter one or more goals.
++
+In the following example, rack awareness and replica capacity are user-provided optimization goals:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec:    
+  goals:
+    - RackAwareGoal,
+    - ReplicaCapacityGoal
+----
+
+. Create or update the resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl apply -f _your-file_
+----
++
+The Cluster Operator requests the optimization proposal from Cruise Control.
+This might take a few minutes depending on the size of the Kafka cluster. 
+
+. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
```

Comment:
This command is wrong. It should be `kubectl get kafkarebalances <rebalance-name> -o jsonpath='{.status}'`

Reply:
Or better yet: `kubectl describe kafkarebalance <rebalance-name>`

Reply:
You can get the exact status output with `kubectl get kafkarebalances <rebalance-name> -o jsonpath='{.status.conditions[0].status}'` but there is no absolute guarantee that the correct condition will always be at position zero.